src/librustc_typeck/check/callee.rs

```diff
@@ -155,6 +155,7 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         }
 
         self.try_overloaded_call_traits(call_expr, adjusted_ty, Some(arg_exprs))
+            .or_else(|| self.try_overloaded_call_traits(call_expr, adjusted_ty, None))
```

Comment:
What's going on here? Can we add a comment explaining this?

Reply:
This is used to detect an uncallable implementation of fn traits, for diagnostic purposes. I will push a comment soon.